zooniverse/front-end-monorepo
# Clean up ZooHeader defaultProps and standard horizontal padding

Includes https://github.com/zooniverse/front-end-monorepo/pull/5659 (merged). Please see that PR for details about `useHasMounted` as a technique for hydration mismatch.

## Package
lib-react-components

## Linked Issue and/or Talk Post

Padding Discussion: https://github.com/zooniverse/front-end-monorepo/discussions/5598
Toward: https://github.com/zooniverse/front-end-monorepo/issues/5630
Somewhat related to https://github.com/zooniverse/front-end-monorepo/issues/5273 (I left a comment over there)

## Describe your changes
- Adjust styling in ZooHeader to match the standardized 30px horizontal padding on desktop and 20px horizontal padding for mobile (Grommet's `medium` value)
- NarrowMainNavMenu component does not care whether there's a user or not, so I moved it to ZooHeader rather than duplicate in SignedOutUserNavigation and SignedInUserNavigation
- SignedOutUserNavigation does not care about user info, so I removed the unused user props from that component
- Remove `defaultProps` from all ZooHeader components
- Removed an old enzyme file from UserMenu
- Removed "narrow" stories from storybook. Those story components were limiting width via Box, but that doesn't communicated to Grommet that the viewport is narrow, so the padding wasn't adjusting as expected. Instead, you can view various ZooHeader stories and use Storybook's built-in viewport adjuster in the top navbar.

## How to Review

- Run lib-react-component's Storybook to see signed in versus signed out states of ZooHeader
- View any route on app-project, app-content-pages, or app-root to see the ZooHeader in the wild

# Checklist
_PR Creator - Please cater the checklist to fit the review needed for your code changes._
_PR Reviewer - Use the checklist during your review. Each point should be checkmarked or discussed before PR approval._

## General

## General UX
Example Staging Project: [i-fancy-cats](https://local.zooniverse.org:3000/projects/brooke/i-fancy-cats)
  - Can be used with a screen reader [BBC guide to VoiceOver](https://bbc.github.io/accessibility-news-and-you/assistive-technology/testing-steps/voiceover-mac.html)
  - Can be used from the keyboard [WebAIM guide to keyboard testing](https://webaim.org/techniques/keyboard/#testing)
  - It is passing accessibility checks in the storybook

## Refactoring
